### PR TITLE
hide members where possible

### PR DESCRIPTION
**Pull request by andrewkolos**
I was poking around in logger.dart when I noticed a few classes have unnecessary exposed member variables. This PR reduces visibility of these in the pursuit of making these classes slightly easier to grok.

<details>

<summary> Pre-launch checklist </summary> 

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

</details>

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 1]
```diff
@@ -1145,27 +1145,28 @@ typedef SlowWarningCallback = String Function();
...
-  final Duration? timeout;
+  final VoidCallback? _onFinish;
+  final Duration? _timeout;
 
   @protected
   final Stopwatch _stopwatch;
 
   @protected
-  String get elapsedTime {
+  String get _elapsedTime {
```
Comment (bkonyi) [inline packages/flutter_tools/lib/src/base/logger.dart:1162]: Does this still need to be marked as `@protected` (here and elsewhere where we have the annotation on library private members)?

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 2]
```diff
@@ -1333,10 +1338,11 @@ class AnonymousSpinnerStatus extends Status {
 
   final List<String> _animation;
 
-  Timer? timer;
+  Timer? _timer;
+  @visibleForTesting
```
Comment (bkonyi) [inline packages/flutter_tools/lib/src/base/logger.dart:1342]: Do we set `ticks` in tests anywhere? If not, should this be a getter for a private field?

**Review by bkonyi — APPROVED** [review]
Nice cleanup! LGTM overall with a couple of small comments.

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 1 above)
Comment (andrewkolos) [reply packages/flutter_tools/lib/src/base/logger.dart:1160]: Thanks for catching this! Fixed in [make ticks readonly, remove unnecessary protected annotations](https://github.com/flutter/flutter/pull/158492/commits/d81d598225749f37da3db436a6c6e34de00d430d)

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 2 above)
Comment (andrewkolos) [reply packages/flutter_tools/lib/src/base/logger.dart:1342]: Fixed in [make ticks readonly, remove unnecessary protected annotations](https://github.com/flutter/flutter/pull/158492/commits/d81d598225749f37da3db436a6c6e34de00d430d)

**Pull request merged**